x-pack/plugins/ml/common/types/capabilities.ts

```diff
@@ -182,3 +198,50 @@ export interface MlCapabilitiesResponse {
 }
 
 export type ResolveMlCapabilities = (request: KibanaRequest) => Promise<MlCapabilities | null>;
+
+interface FeatureCapabilities {
+  ad: MlCapabilitiesKey[];
+  dfa: MlCapabilitiesKey[];
+  nlp: MlCapabilitiesKey[];
```

Comment:
should it be different capabilities for `ad`, `dfa` and `npl`?  
You can extract the type from the `featureCapabilities` constant below.

Reply:
The interface here is only used for `featureCapabilities` below, it is not exported out. So it's a bit of a chicken an egg situation.